pkg/controller/releasemanager/plan/syncApp.go

```diff
@@ -97,106 +96,152 @@ func (p *SyncApp) serviceHost() string {
 	return fmt.Sprintf("%s.%s.svc.cluster.local", p.App, p.Namespace)
 }
 
-func (p *SyncApp) releaseMatches(log logr.Logger, clusterName string, port picchuv1alpha1.PortInfo) []*istio.HTTPMatchRequest {
-	matches := []*istio.HTTPMatchRequest{{
-		Port:     uint32(port.Port),
-		Gateways: []string{"mesh"},
-		Uri:      &istio.StringMatch{MatchType: &istio.StringMatch_Prefix{Prefix: "/"}},
-	}}
+func (p *SyncApp) publicHosts(port picchuv1alpha1.PortInfo, options plan.Options) []string {
+	return p.ingressHosts(picchuv1alpha1.PortPublic, port, options.DefaultDomains.Public)
+}
 
-	portNumber := uint32(port.Port)
-	gateways := []string{"mesh"}
-	hosts := port.Hosts
-
-	switch port.Mode {
-	case picchuv1alpha1.PortPublic:
-		if p.PublicGateway == "" {
-			log.Info("publicGateway undefined in Cluster for port", "port", port)
-			return matches
-		}
-		gateways = []string{p.PublicGateway}
-		portNumber = uint32(port.IngressPort)
-	case picchuv1alpha1.PortPrivate:
-		if p.PrivateGateway == "" {
-			log.Info("privateGateway undefined in Cluster for port", "port", port)
-			return matches
-		}
-		gateways = []string{p.PrivateGateway}
-
-		defaultDomainsMap := map[string]bool{}
-		for _, defaultDomain := range p.DefaultDomains {
-			defaultDomainsMap[defaultDomain] = true
-			if clusterName != "" {
-				clusterDomain := fmt.Sprintf("%s.cluster.%s", clusterName, defaultDomain)
-				defaultDomainsMap[clusterDomain] = true
-			}
-		}
-		for domain := range defaultDomainsMap {
-			hosts = append(hosts, fmt.Sprintf("%s.%s", p.Namespace, domain))
-			if port.Name != "" {
-				hosts = append(hosts, fmt.Sprintf("%s-%s.%s", p.Namespace, port.Name, domain))
-			}
-		}
+func (p *SyncApp) privateHosts(port picchuv1alpha1.PortInfo, options plan.Options) []string {
+	return p.ingressHosts(picchuv1alpha1.PortPrivate, port, options.DefaultDomains.Private)
+}
 
-		portNumber = uint32(port.IngressPort)
+func (p *SyncApp) authorityMatch(hosts []string) *istio.StringMatch {
+	sort.Strings(hosts)
+	regex := fmt.Sprintf("^(%s)", strings.Join(hosts, "|"))
```

Comment:
```suggestion
	hostPatterns := make([]string, 0, len(hosts))
	for _, host := range hosts {
		hostPatterns = append(hostPatterns, regexp.QuoteMeta(host))
	}
	sort.Strings(hostPatterns)
	regex := fmt.Sprintf(`^(%s)(:[0-9]+)$`, strings.Join(hostPatterns, "|"))
```
this will change (e.g.) `medium.com` to `medium\.com` in the pattern. also iirc the only reason we used a prefix match before was because (sometimes?) the HTTP host would include a port number; if we're doing a regex, let's try to match the whole host